cdk/domino_cdk/util.py

```diff
@@ -146,3 +153,60 @@ def overlay(alpha: dict, omega: dict) -> dict:
             return {}
         base_dict = check_type(dictionaries[0])
         return base_dict if len(dictionaries) == 1 else overlay(base_dict, cls.deep_merge(*dictionaries[1:]))
+
+    @staticmethod
+    def create_lambda(
+        scope: cdk.Construct,
+        stack_name: str,
+        dirname: str,
+        name: str,
+        environment: Dict[str, str],
+        resources: List[str],
+        actions: List[str],
+    ) -> cdk.Construct:
+        with open(path.join(dirname, "lambda", name + ".py"), encoding="utf-8") as fp:
+            on_event_code_body = fp.read()
+        on_event = lambda_.Function(
+            scope,
+            name + "_on_event",
+            function_name=f"{stack_name}-{name}",
+            runtime=lambda_.Runtime.PYTHON_3_7,
+            handler="index.on_event",
+            code=lambda_.InlineCode(on_event_code_body),
+            environment=environment,
+            timeout=cdk.Duration.seconds(600),  # default is 3 seconds
+            log_retention=logs.RetentionDays.ONE_DAY,  # defaults to never delete logs
+        )
+        statement = iam.PolicyStatement()
+        for r in resources:
+            statement.add_resources(r)
+        for a in actions:
+            statement.add_actions(a)
+        on_event.add_to_role_policy(statement)
+
+        is_complete = None
+        try:
+            with open(path.join(dirname, "lambda", name + "_is_complete.py"), encoding="utf-8") as fp:
```

Comment:
This is an honest question, not a leading one, but could this just be rolled into main lambda code so we don't have to look for two files? Like if you provide the same lambda for both things, and `is_complete` is a function in the code, will it just do the right thing? Also, do we anticipate having a use case for `is_complete` any time soon? ATM it looks like it's a code path that won't get exercised, but could imagine there's something coming up for it.

Reply:
1. No, Provider class requires a separate lambda resource for is_complete. The only way to share the code is to create a bucket resource for the code and use it in 2 lambdas but this is 3 resources instead of 2.
2. Is complete is used to avoid his 10 minutes timeout for lambda. This way you have 2 functions, each completes in 3 seconds and doesn't use your container engine with the corresponding costs.
3. I delete Provider and is_complete because we really do not care.